src/components/pages/home.jsx

```diff
@@ -32,28 +34,32 @@ const HomePage = React.createClass({
     return {inputValue: ""}
   },
   handleClearSearchClicked() {
-    this.setState({inputValue: ""})
-    this.context.router.push("")
+    const query = ""
+    this.setState({inputValue: query})
+    this.context.setQuery(query)
+    this.context.router.push(query)
```

Comment:
This should probably be refactored in a call that handles both clearing and setting the input to avoid duplication and inconsistent behaviour (which is already the case with `scrollIntoView` only called on input change, not on clearing).